mmv1/products/datapipeline/DataPipeline.yaml

```
@@ -0,0 +1,400 @@
+                      - :FLEXRS_COST_OPTIMIZED
+              - !ruby/object:Api::Type::Boolean
+                name: 'update'
+                description: |
+                  Set this to true if you are sending a request to update a running streaming job. When set, the job name should be the same as the running job.
+              - !ruby/object:Api::Type::KeyValuePairs
+                name: 'transformNameMappings'
+                description: |
+                  'Use this to pass transform name mappings for streaming update jobs. Example: {"oldTransformName":"newTransformName",...}'
+                  'An object containing a list of "key": value pairs. Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.'
+              - !ruby/object:Api::Type::String
+                name: 'containerSpecGcsPath'
+                description: |
+                  Cloud Storage path to a file with a JSON-serialized ContainerSpec as content.
+          - !ruby/object:Api::Type::String
+            name: 'location'
+            description: |
+              The regional endpoint to which to direct the request. For example, us-central1, us-west1.
+            required: true
+          - !ruby/object:Api::Type::Boolean
+            name: 'validateOnly'
+            description: |
+              If true, the request is validated but not actually executed. Defaults to false.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'scheduleInfo'
+    description: |
+      Internal scheduling information for a pipeline. If this information is provided, periodic jobs will be created per the schedule. If not, users are responsible for creating jobs externally.
+      https://cloud.google.com/dataflow/docs/reference/data-pipelines/rest/v1/projects.locations.pipelines#schedulespec
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'schedule'
+        description: |
+          Unix-cron format of the schedule. This information is retrieved from the linked Cloud Scheduler.
+      - !ruby/object:Api::Type::String
+        name: 'timeZone'
+        description: |
+          Timezone ID. This matches the timezone IDs used by the Cloud Scheduler API. If empty, UTC time is assumed.
+        default_value: UTC
+      - !ruby/object:Api::Type::String
+        name: 'nextJobTime'
+        description: |
+          When the next Scheduler job is going to run.
+          A timestamp in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to nine fractional digits. Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+        output: true
+  - !ruby/object:Api::Type::Integer
+    name: 'jobCount'
+    description: |
+      Number of jobs.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'schedulerServiceAccountEmail'
+    description: |
+      Optional. A service account email to be used with the Cloud Scheduler job. If not specified, the default compute engine service account will be used.
```

```suggestion
  - !ruby/object:Api::Type::String
    name: 'schedulerServiceAccountEmail'
    description: |
      Optional. A service account email to be used with the Cloud Scheduler job. If not specified, the default compute engine service account will be used.
    immutable: true
```

Adding immutable since we can't read this value from the api